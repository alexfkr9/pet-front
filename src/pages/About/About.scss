@import "/src/styles/variables";

.container:has(.sniff-about-info) {
  padding: 0;
}

.sniff-about-section {
  display: flex;
  flex-direction: column;
}

.sniff-about-info {
  background-color: $--color-secondary;
  padding: 40px 28px;
  text-align: left;
}

.sniff-about-info-title {
  font-weight: 800;
  font-size: 30px;
  margin-bottom: $--space-24;
}

.sniff-about-info-text {
  font-weight: 500;
  line-height: 1.5;
  font-size: $--font-size-base;

  & :last-child {
    margin: 0;
  }
}

.sniff-about-item-image {
  display: none;
}

.sniff-about-text {
  border-top: $--base-border;
  background-color: $--green;
  text-align: center;
  padding: 60px 0px;
}

.sniff-about-text-title {
  font-weight: 800;
  margin-bottom: $--space-24;
}

.sniff-about-text-btn {
  border: $--btn-outline-active-bg;
  left: 0;
  top: 0;
  padding: $--space-12 $--space-24;
  font-size: $--font-size-base;

  &:hover {
    left: -4px;
    top: -4px;
  }
}

@media (min-width: $--md) {
  .sniff-about-section {
    flex-direction: row;
  }

  .sniff-about-info {
    width: 60%;
    border-right: $--base-border;
  }

  .sniff-about-info-title {
    font-size: 40px;
  }

  .sniff-about-info-text {
    font-size: $--font-size-medium;
  }

  .sniff-about-item {
    width: 40%;
  }

  .sniff-about-item-image {
    display: block;
    object-fit: cover;
    width: 100%;
    height: 100%;
  }

  .sniff-about-text {
    padding: 71px 0px;
  }

  .sniff-about-text-title {
    font-size: 40px;
    margin-bottom: 30px;
  }

  .sniff-about-text-btn {
    padding: 24px 48px;
    font-size: 20px;
  }
}

@media (min-width: $--lg) {
  .sniff-about-info {
    padding: 80px;
    width: 50%;
  }

  .sniff-about-item {
    width: 50%;
  }

  .sniff-about-text-title {
    margin-bottom: $--space-48;
    font-size: 56px;
    line-height: 1.1;
  }
}

@media (min-width: 1440px) {
  .container:has(.sniff-about-info) {
    min-width: 1440px;
  }
}
