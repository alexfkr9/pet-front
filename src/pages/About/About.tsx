import "./About.scss";
import Pattern from "../../assets/images/Pattern.svg";
import Button from "react-bootstrap/Button";

const About = () => {
  // eslint-disable-next-line no-lone-blocks
  {
    /* useEffect(() => {
    userApi
      .getUserById(7, {
        headers: new AxiosHeaders({
          Authorization: "Basic c3N1Y2hrb3ZAZ21haWwuY29tOmJlc3RtZW50b3I="
        })
      })
      .then((response: any) => {
        // eslint-disable-next-line no-console
        console.log(response.data);
      })
      .catch((err: any) => {
        // eslint-disable-next-line no-console
        console.log(err);
      });
  }, []);
*/
  }
  return (
    <section className="container">
      <div className="sniff-about-section">
        <div className="sniff-about-info">
          <h2 className="sniff-about-info-title">Хто ми такі?</h2>
          <div className="sniff-about-info-text">
            <p>
              Натхненна численними історiями про те, як українці допомагають
              чотирилапим, навіть ризикуючи власним життям та безпекою, команда
              Sniff допомагає обʼєднатись всім, хто розшукує свого улюбленця або
              готовий надати притулок тваринці, про яку зараз немає кому
              подбати.
            </p>
            <p>
              Наша платформа охоплює користувачів з усіх регіонів України: це і
              волонтерські організаціі, і притулки, а також звичайні люди з
              надзвичайно добрим серцем.
            </p>
          </div>
        </div>

        <div className="sniff-about-item">
          <img className="sniff-about-item-image" src={Pattern} alt="Pattern" />
        </div>
      </div>
      <div className="sniff-about-text">
        <h1 className="sniff-about-text-title">
          Маєте питання чи <br /> пропозиції?
        </h1>
        <Button className="sniff-about-text-btn" href="#">
          Написати нам
        </Button>
      </div>
    </section>
  );
};

export default About;
