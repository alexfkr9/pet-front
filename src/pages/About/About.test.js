import { getByText, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import About from "./About";

describe("About component", () => {
  it("should render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <About />
      </MemoryRouter>
    );

    const html = asFragment();

    expect(html).toMatchSnapshot();
  });
});
