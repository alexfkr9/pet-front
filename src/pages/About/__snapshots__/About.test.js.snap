// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should render 1`] = `
<DocumentFragment>
  <section>
    <div
      class="sniff-about-section"
    >
      <div
        class="sniff-about-info"
      >
        <h2
          class="sniff-about-info-title"
        >
          Хто ми такі?
        </h2>
        <p
          class="sniff-about-info-text"
        >
          Натхненна численними історiями про те, як українці допомагають чотирилапим, навіть ризикуючи власним життям та безпекою, команда Sniff допомагає обʼєднатись всім, хто розшукує свого улюбленця або готовий надати притулок тваринці, про яку зараз немає кому подбати.
        </p>
        <p>
          Наша платформа охоплює користувачів з усіх регіонів України: це і волонтерські організаціі, і притулки, а також звичайні люди з надзвичайно добрим серцем.
        </p>
        <p />
      </div>
      <div
        class="sniff-about-item"
      >
        <img
          alt="Pattern"
          class="sniff-about-item-image"
          src="Pattern.svg"
        />
      </div>
    </div>
    <div
      class="sniff-about-text"
    >
      <h1
        class="sniff-about-text-title"
      >
        Маєте питання чи 
        <br />
         пропозиції?
      </h1>
      <a
        class="sniff-about-text-btn btn btn-primary"
        href="#"
        role="button"
        tabindex="0"
      >
        Написати нам
      </a>
    </div>
  </section>
</DocumentFragment>
`;
